File: lib/sheaf/ns.ex
defmodule Sheaf.NS do
  @moduledoc """
  RDF vocabularies used by Sheaf.
  """

  use RDF.Vocabulary.Namespace
  require RDF.Turtle

  defvocab(DOC,
    base_iri: "https://less.rest/sheaf/",
    file: "../sheaf-schema.ttl"
  )

  defvocab(AS,
    base_iri: "https://www.w3.org/ns/activitystreams#",
    file: "../activitystreams2.ttl"
  )

  defvocab(CSVW,
    base_iri: "http://www.w3.org/ns/csvw#",
    terms: ~w[
      Column
      Schema
      Table
      TableGroup
      column
      datatype
      name
      table
      tableSchema
      title
    ]
  )

  defvocab(DCAT,
    base_iri: "http://www.w3.org/ns/dcat#",
    terms: ~w[
      Dataset
      Distribution
      byteSize
      distribution
      mediaType
    ]
  )

  defvocab(PROV,
    base_iri: "http://www.w3.org/ns/prov#",
    file: "../prov-o.ttl"
  )

  defvocab(BFO,
    base_iri: "https://node.town/bfo#",
    file: "../sheaf-ext.ttl"
  )

  defvocab(BIBO,
    base_iri: "http://purl.org/ontology/bibo/",
    terms: ~w[
      Journal
      doi
      degree
      isbn
      issn
      numPages
      pageEnd
      pageStart
      volume
    ]
  )

  defvocab(BIRO,
    base_iri: "http://purl.org/spar/biro/",
    terms: ~w[
      BibliographicReference
      ReferenceList
      references
    ]
  )

  defvocab(CITO,
    base_iri: "http://purl.org/spar/cito/",
    terms: ~w[
      cites
    ]
  )

  defvocab(CO,
    base_iri: "http://purl.org/co/",
    terms: ~w[
      Collection
      Set
      element
      elementOf
    ]
  )

  defvocab(DCTERMS,
    base_iri: "http://purl.org/dc/terms/",
    terms: ~w[
      creator
      date
      format
      identifier
      isPartOf
      publisher
      title
    ]
  )

  defvocab(DEO,
    base_iri: "http://purl.org/spar/deo/",
    terms: ~w[
      BibliographicReference
    ]
  )

  defvocab(DOI,
    base_iri: "http://dx.doi.org/",
    terms: [],
    strict: false
  )

  defvocab(FABIO,
    base_iri: "http://purl.org/spar/fabio/",
    terms: ~w[
      Book
      BookChapter
      ComputerFile
      DoctoralThesis
      JournalArticle
      PositionPaper
      ResearchPaper
      ScholarlyWork
      hasDOI
      hasIssueIdentifier
      hasPageRange
      hasPublicationYear
      hasVolumeIdentifier
      isPortrayalOf
      isRepresentationOf
    ]
  )

  defvocab(FOAF,
    base_iri: "http://xmlns.com/foaf/0.1/",
    terms: ~w[
      Organization
      Person
      familyName
      givenName
      name
    ]
  )

  defvocab(FRBR,
    base_iri: "http://purl.org/vocab/frbr/core#",
    terms: ~w[
      exemplarOf
      partOf
      realizationOf
    ]
  )

  defvocab(PRISM,
    base_iri: "http://prismstandard.org/namespaces/basic/2.1/",
    terms: ~w[
      doi
      endingPage
      issn
      issueIdentifier
      startingPage
      volume
    ]
  )
end

File: tools/sheaf-client/src/ns.ts
export const RDF = {
  type: "http://www.w3.org/1999/02/22-rdf-syntax-ns#type",
  first: "http://www.w3.org/1999/02/22-rdf-syntax-ns#first",
  rest: "http://www.w3.org/1999/02/22-rdf-syntax-ns#rest",
  nil: "http://www.w3.org/1999/02/22-rdf-syntax-ns#nil",
}

export const RDFS = {
  label: "http://www.w3.org/2000/01/rdf-schema#label",
}

export const SHEAF = {
  base: "https://less.rest/sheaf/",
  resourceBase: "https://sheaf.less.rest/",
  metadataGraph: "https://less.rest/sheaf/metadata",
  Document: "https://less.rest/sheaf/Document",
  Thesis: "https://less.rest/sheaf/Thesis",
  Paper: "https://less.rest/sheaf/Paper",
  Transcript: "https://less.rest/sheaf/Transcript",
  Spreadsheet: "https://less.rest/sheaf/Spreadsheet",
  Section: "https://less.rest/sheaf/Section",
  ParagraphBlock: "https://less.rest/sheaf/ParagraphBlock",
  ExtractedBlock: "https://less.rest/sheaf/ExtractedBlock",
  Row: "https://less.rest/sheaf/Row",
  children: "https://less.rest/sheaf/children",
  paragraph: "https://less.rest/sheaf/paragraph",
  text: "https://less.rest/sheaf/text",
  sourceHtml: "https://less.rest/sheaf/sourceHtml",
  sourcePage: "https://less.rest/sheaf/sourcePage",
  sourceKey: "https://less.rest/sheaf/sourceKey",
}

export const DCTERMS = {
  creator: "http://purl.org/dc/terms/creator",
  isPartOf: "http://purl.org/dc/terms/isPartOf",
  publisher: "http://purl.org/dc/terms/publisher",
  title: "http://purl.org/dc/terms/title",
}

export const BIBO = {
  numPages: "http://purl.org/ontology/bibo/numPages",
  status: "http://purl.org/ontology/bibo/status",
}

export const CITO = {
  cites: "http://purl.org/spar/cito/cites",
}

export const FABIO = {
  Book: "http://purl.org/spar/fabio/Book",
  BookChapter: "http://purl.org/spar/fabio/BookChapter",
  DoctoralThesis: "http://purl.org/spar/fabio/DoctoralThesis",
  JournalArticle: "http://purl.org/spar/fabio/JournalArticle",
  ScholarlyWork: "http://purl.org/spar/fabio/ScholarlyWork",
  hasIssueIdentifier: "http://purl.org/spar/fabio/hasIssueIdentifier",
  hasPageRange: "http://purl.org/spar/fabio/hasPageRange",
  hasPublicationYear: "http://purl.org/spar/fabio/hasPublicationYear",
  hasVolumeIdentifier: "http://purl.org/spar/fabio/hasVolumeIdentifier",
  isRepresentationOf: "http://purl.org/spar/fabio/isRepresentationOf",
}

export const FOAF = {
  name: "http://xmlns.com/foaf/0.1/name",
}
